src/compiler/utilities.ts

```diff
@@ -3939,8 +3939,9 @@ namespace ts {
         let hasTrailingComment = false;
 
         function updateLineCountAndPosFor(s: string) {
-            const lineStartsOfS = computeLineStarts(s);
-            if (lineStartsOfS.length > 1) {
+            // Most arguments don't contain one of JS's four linebreak characters
+            if (/[\r\n\u2028\u2029]/.test(s)) {

```

Comment:
Drive by comment 🚗: I wonder if it could be beneficial to:

1. Create the regex once
2. Leverage the regex's `lastIndex` instead of using `computeLineStarts`. This would involve iteratively matching to count the lines, then using the last `lastIndex` +1 (and probably +2 for u2028/u2029) in favor of `last(lineStartsOfS)`.

Btw, `lineStart = (linePos - output.length) === 0;` a couple lines lower can just be `lineStart = linePos === output.length;` (probably won't make a difference).

Reply:
Thanks!  If I don't abandon this in favor of https://github.com/microsoft/TypeScript/pull/44242, I'll give that a shot (and, yeah, the `=== 0` was bugging me too).